src/chess/piece/Piece.java

```diff
@@ -18,17 +18,19 @@ public Piece(Color color, Map<Color, Character> symbols) {
 
     this.color = color;
     symbol = s;
+    this.lastMovedTurn = 0;
   }
 
-  public boolean isValidMove(Position newPosition) {
-    if (newPosition.getRow() > 0
-        && newPosition.getCol() > 0
-        && newPosition.getRow() < 8
-        && newPosition.getCol() < 8) {
-      return true;
-    } else {
-      return false;
-    }
+  public Color getColor() {
+    return color;
+  }
+
+  public void setLastMovedTurn(int lastMovedTurn) {
+    this.lastMovedTurn = lastMovedTurn;
+  }
+
+  public boolean isValidMove(int x, int y, boolean isEnemyExisted) {
+    return true;
```

Comment:
This doesn't make sense!
You can just define this as an abstract method without any content
So you don't need to call this as `super.isvalidMove()` from children